atlasdb-feign/src/main/java/com/palantir/atlasdb/http/PollingRefreshable.java

```diff
@@ -46,50 +49,58 @@
     private final Duration refreshInterval;
     private final ScheduledExecutorService poller;
 
-    private final Refreshable<T> refreshable = Refreshable.empty();
+    private final Function<T, U> transform;
+    private final Refreshable<U> refreshable = Refreshable.empty();
 
-    private T lastSeenValue;
+    private T lastSeenValue = null;
 
     private PollingRefreshable(Supplier<T> supplier,
             Duration refreshInterval,
+            Function<T, U> transform,
             ScheduledExecutorService poller) {
         Preconditions.checkArgument(!refreshInterval.isNegative() && !refreshInterval.isZero(),
                 "Refresh interval must be positive, but found %s", refreshInterval);
 
         this.supplier = supplier;
         this.refreshInterval = refreshInterval;
+        this.transform = transform;
         this.poller = poller;
 
         try {
             lastSeenValue = supplier.get();
-            refreshable.set(lastSeenValue);
+            refreshable.set(transform.apply(lastSeenValue));
         } catch (Exception e) {
             log.info("Exception occurred in supplier when trying to populate the initial value.");
```

Comment:
nit: This is no longer correct (we might get an exception that was in the transform, not in the supplier). Also, probably best to include what the exception was.